File: week06/29/sumOfLeftLeaves.py
def sumOfLeftLeaves(root):
    def dft(node):
        if node is None:
            return 0
        sum = 0
        if node.left and node.left.left is None and node.left.right is None:  # 자식 없을 때
            sum += node.left.val
        sum += dft(node.left)
        sum += dft(node.right)

        return sum
    
    result = dft(root)
        
    return result

File: week06/25/SumOfLeftLeaves.c
#include <stdio.h>
#include <stdlib.h>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

int sumOfLeftLeaves(struct TreeNode* root) {
    int result = 0;
    // base case(1) : no child
    if (root->left == NULL || root->right == NULL) {
        return 0;
    }
    // recursion about right child
    if (root->right) {result += sumOfLeftLeaves(root->right);}

    if (root->left->left) {result += sumOfLeftLeaves(root->left);}
    if (root->left->right) {result += sumOfLeftLeaves(root->right);}
    // base case(2) : target found
    if ((!root->left->left) && (!root->left->right)) {result += root->left->val;}
    
    return result;
}